src/main/java/it/geosolutions/httpproxy/HTTPProxy.java

```diff
@@ -209,13 +215,15 @@ private Boolean isNonProxyHost(String host) {
             if (nonProxyHostProp.endsWith("\"")) {
                 nonProxyHostProp = nonProxyHostProp.substring(0, nonProxyHostProp.length() - 1);
             }
+            LOGGER.info("http.nonProxyHosts: " + nonProxyHostProp);
             StringTokenizer tokenizer = new StringTokenizer(nonProxyHostProp, "|");
             while (tokenizer.hasMoreTokens()) {
                 String str = tokenizer.nextToken().trim();
-                str = str.replace("*", "\\w*");
+                str = str.replace("*", "[\\w-]*");
```

Comment:
The parsing of non should only be done once, do you confirm is the case?

Secondly, is the comparison case sensitive? Please double check with Oracle spec.

Reply:
1. I could see that this parsing is done for every request.
2. There is no explicit mention of case sensitivity on oracle specs. It refers to matching the ```patterns ``` which means that the comparison is case sensitive.